### DISCUSSION TITLE
Jetbrains IDE autocompletion

### DISCUSSION THREAD

**Discussion by RemyMachado** [chakra-ui/panda#1671] [General]
Hello, I'm currently using Webstorm with Pandacss and I have no autocompletion for the css strings on the recipe side.
I saw a plugin on vscode, but nothing for jetbrains, any suggestion?

Here's an example:

```javascript
{
        ...
        borderColor: 'fg.onSurface', // no autocomplete
        ...
 }
```

**Comment by anubra266** [comment]
@RemyMachado The extension is not required for autocomplete. It's typescript.

**Reply by RemyMachado** [reply]
I'm not sure I follow. How do I get suggestions for `onSurface` when I start typing `"fg.`?

**Reply by anubra266** [reply]
This is example in online playground. No extension. The styled system generated by panda css, generates the colors in the types.

![CleanShot 2023-11-12 at 21 10 51@2x](https://github.com/chakra-ui/panda/assets/30869823/50c26335-d925-49d5-a7c8-c38811446ffb)

**Reply by RemyMachado** [reply]
I'm confused. I just checked with coworkers using VSCode and they get the autocompletion from the plugin. When they disable it, they don't get the autocompletion no more...

**Reply by anubra266** [reply]
@RemyMachado The extension can provide superpowered suggestions. e.g. a preview of color tokens, or the rem and px values of sizes and font sizes.
But without the extension, you get raw suggestions from typescript like the one in my screenshot.
Maybe I'll install webstorm and try.

**Reply by anubra266** [reply]
@RemyMachado I can get the suggestions in Webstorm
![CleanShot 2023-11-14 at 14 32 54@2x](https://github.com/chakra-ui/panda/assets/30869823/82b648e6-101c-4aae-8e12-9e1ac5b9b197)

**Reply by RemyMachado** [reply]
Yes, I'm interested in custom tokens my designer team defined. I'm not quite recognizing what's custom and what's not yet.

Thanks for trying it on Webstorm.

**Comment by Mydayyy** [comment]
> @RemyMachado I can get the suggestions in Webstorm ![CleanShot 2023-11-14 at 14 32 54@2x](https://private-user-images.githubusercontent.com/30869823/282790432-82b648e6-101c-4aae-8e12-9e1ac5b9b197.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE2MTM0NjcsIm5iZiI6MTcwMTYxMzE2NywicGF0aCI6Ii8zMDg2OTgyMy8yODI3OTA0MzItODJiNjQ4ZTYtMTAxYy00YWFlLThlMTItOWUxYWM1YjliMTk3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjAzVDE0MTkyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM5OGRjN2M4YjNiMjlkNzE2MTY3YzA1ZjJjNDQxYTU3MjEyOTZiMzRhNGZkOTk0YjUyNGNmMDBhNWE0ZjhmNmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rWDzFiS9TlnvdtCV1NfRnDUped4eaCXaEwF-9Rco7cY)

Do you have any special configuration or plugins to get autocomplete here? My TS completion in general is working in webstorm but I get nothing for pandacss generated content.

E.g `<Grid flex="">`  here, neither attributes for Grid (like flex) nor any values for the attributes have suggestions

Edit: I do have some completion now after invalidating all caches (which I already did in the past but it didnt help). But it still feels really bumpy and slow compared to vscode. Might just have to drop webstorm for this

**Reply by anubra266** [reply]
No...I just installed webstorm and opened a panda css project

**Reply by Mydayyy** [reply]
It seems I am not the only one with the issue using webstorm. Webstorm appears to ignore the styled-system directory for types completely and always links back to the generic types in node_modules (although tsconfig is configured as documented). I believe this is most likely a webstorm issue - and unfortunately they are extremely slow with fixing bugs.

It's still not working correctly with the latest version that just got released

**Reply by Yuripetusko** [reply]
For me it works out of the box, but it takes a while when panda is compiling on change. It looks like typescript autocompletion/indexing is blocked while Panda is compiling. And I am on mac mini m2 with 32gb ram. I posted about this here too
https://github.com/chakra-ui/panda/discussions/1937

**Reply by Yuripetusko** [reply]
also @Mydayyy make sure you add `"styled-system",` to `include` array in your `tsconfig.json`

**Reply by aloysb** [reply]
Same issue here, I'm using the postcss plugin.

No issue in VSCode, but Webstorm is playing up
